metadata-ingestion/src/datahub/ingestion/source/sql/presto.py

```diff
@@ -0,0 +1,136 @@
+from textwrap import dedent
+
+from pydantic.fields import Field
+from pyhive.sqlalchemy_presto import PrestoDialect
+from sqlalchemy import exc, sql
+from sqlalchemy.engine import reflection
+
+from datahub.ingestion.api.decorators import (
+    SourceCapability,
+    SupportStatus,
+    capability,
+    config_class,
+    platform_name,
+    support_status,
+)
+from datahub.ingestion.source.sql.trino import (
+    TrinoConfig,
+    TrinoSource,
+    _get_columns,
+    get_table_names,
+)
+
+
+# On Presto the information_schema.views does not return views but the tables table returns
+@reflection.cache  # type: ignore
+def get_view_names(self, connection, schema: str = None, **kw):  # type: ignore
+    schema = schema or self._get_default_schema_name(connection)
+    if schema is None:
+        raise exc.NoSuchTableError("schema is required")
+    query = dedent(
+        """
+        SELECT "table_name"
+        FROM "information_schema"."tables"
+        WHERE "table_schema" = :schema and "table_type" = 'VIEW'
+    """
+    ).strip()
+    res = connection.execute(sql.text(query), schema=schema)
+    return [row.table_name for row in res]
+
+
+# The pyhive presto driver doesn't return view definitions, so we have to query it
+@reflection.cache  # type: ignore
+def get_view_definition(self, connection, view_name, schema=None, **kw):
+    schema = schema or self._get_default_schema_name(connection)
+    if schema is None:
+        raise exc.NoSuchTableError("schema is required")
+
+    if view_name is None:
+        raise exc.NoSuchTableError("view_name is required")
+
+    query = dedent(
+        f"""
+        SHOW CREATE VIEW "{schema}"."{view_name}"
+    """
+    ).strip()
+    res = connection.execute(sql.text(query))
+    return next(res)[0]
+
+
+@reflection.cache  # type: ignore
+def get_table_comment(self, connection, table_name: str, schema: str = None, **kw):  # type: ignore
```

Comment:
this looks mostly duplicated from the trino file - any way to reduce the duplication?